### PR TITLE
Frame of reference UID fix

### PR DESCRIPTION
**Pull request by michaelkain**
This fix is the follow up of an issue raised by GE.
With the fix respect the DICOM structure concerning the frameOfReferenceUID, as given in the original DICOM.

### CODE REVIEW

**Comment by michaelkain** [comment]
Hi, I tested it successfully with my local ShUp. Might be worth testing on the server too. Cheers, Michael

File: shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java [hunk 1]
```diff
@@ -350,6 +354,25 @@ private void anonymizeSOPInstanceUID(int tagInt, Attributes attributes, String m
...
+			} catch (Exception e) {
+				LOG.error(e.getMessage());
+			}
+			value = newUID;
+			frameOfReferenceUIDs.put(attributes.getString(tagInt), value);
+		}
+		anonymizeTagAccordingToVR(attributes, tagInt, value);
+	}
+
 	private void anonymizeSeriesInstanceUID(int tagInt, Attributes attributes, Map<String, String> seriesInstanceUIDs) {
```
Comment (jcomedouteau) [inline shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java:376]: If I'm not wrong, the three methods could be factorized into a single one (and we just give a different Map<string, string> for each case), but I don't know if it's really relevant in terms of "code clarity" (?)

File: shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java [hunk 2]
```diff
@@ -217,6 +219,8 @@ public void performAnonymization(final File dicomFile, Map<String, String> anony
 						anonymizeSOPInstanceUID(tagInt, datasetAttributes, mediaStorageSOPInstanceUIDGenerated);
 					} else if (tagInt == Tag.SeriesInstanceUID) {
 						anonymizeSeriesInstanceUID(tagInt, datasetAttributes, seriesInstanceUIDs);
+					} else if (tagInt == Tag.FrameOfReferenceUID) {
```
Comment (LaurentPV) [inline shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java:222]: Maybe change "else if" to a switch case.

File: shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java [hunk 3]
```diff
@@ -523,4 +484,25 @@ else if (vr.equals(VR.AE) || vr.equals(VR.AS) || vr.equals(VR.CS) || vr.equals(V
 		// VR.OD = Other Double String
 	}
 
+	private void anonymizeUID(int tagInt, Attributes attributes, Map<String, String> UIDs) {
```
Comment (michaelkain) [inline shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java:487]: @LaurentPV the problem now is with the method factorization, that it produces too much logs
the info that the StudyInstanceUID has been generated, one by exam, is important, but one log
per serie is just too much, probably the reason, why I kept the 3 methods in the first place

Comment (LaurentPV) [reply shanoir-ng-anonymization/src/main/java/org/shanoir/anonymization/anonymization/AnonymizationServiceImpl.java:487]: I added a condition to log only when studyInstanceUID is concerned. If you think that's really better with two different methods I'll rollback.

**Pull request merged**